bowtie/_cli.py

```diff
@@ -1684,6 +1708,623 @@ async def info(
+        BarColumn(bar_width=None),
+        TextColumn("[progress.percentage]{task.percentage:>3.0f}%"),
+        "•",
+        MofNCompleteColumn(),
+        "•",
+        TimeElapsedColumn(),
+        console=console.Console(),
+        transient=True,
+    )
+
+    if versions:
+        total_files = len(versions) * len(list(dialects))
+        actual_downloaded_files = 0
+
+        task = progress.add_task(
+            description=(
+                "Preparing to download and parse versioned "
+                f"reports of {id} for {pretty_names_str}"
+            ),
+            total=total_files,
+        )
+
+        async with httpx.AsyncClient(timeout=10) as client:
+
+            async def download_and_parse_versioned_report_for(
+                version: str,
+                dialect: Dialect,
+            ):
+                try:
+                    url = (
+                        HOMEPAGE
+                        / "implementations"
+                        / id
+                        / f"v{version}"
+                        / f"{dialect.short_name}.json"
+                    )
+                    response = await client.get(str(url))
+                    response.raise_for_status()
+                    progress.update(
+                        task,
+                        description=(
+                            f"Downloading and Parsing: "
+                            f"v{version}/{dialect.short_name}.json"
+                        ),
+                    )
+                except httpx.HTTPStatusError:
+                    report = _report.Report.empty(dialect=dialect)
+                    progress.update(task, advance=1)
+                    return version, dialect, report
+                else:
+                    nonlocal actual_downloaded_files
+                    actual_downloaded_files += 1
+
+                    report = _report.Report.from_serialized(
+                        response.iter_lines(),
+                    )
+                    progress.update(task, advance=1)
+                    return version, dialect, report
+
+            with progress:
+                responses = await asyncio.gather(
+                    *[
+                        download_and_parse_versioned_report_for(
+                            version,
+                            dialect,
+                        )
+                        for version in versions
+                        for dialect in dialects
+                    ],
+                )
+
+                progress.update(
+                    task,
+                    description=(
+                        "Successfully downloaded and parsed all versioned "
+                        f"reports of {id} for {pretty_names_str}!"
+                    ),
+                    completed=actual_downloaded_files,
+                    total=actual_downloaded_files,
+                )
+                return responses
+    else:
+        total_files = len(list(dialects))
```

Comment:
`len(list(...))` is basically always wrong. Just `len`, or else if it's an iterator, you need to make it a list somewhere earlier.

Reply:
```python
                )
                return responses
    else:
        total_files = len(list(dialects))
```

here for the lines you've pointed out, the code is only entering the else block if versions were empty (again something that would get fixed by the other PR as pointed out below). So list(dialects) is only happening once at this place.